jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/BuildImageMojo.java

```diff
@@ -0,0 +1,180 @@
+
+  @Override
+  public void execute() throws MojoExecutionException {
+    SourceFilesConfiguration sourceFilesConfiguration = getSourceFilesConfiguration();
+
+    BuildConfiguration buildConfiguration =
+        BuildConfiguration.builder()
+            .setBaseImageServerUrl(baseImageRegistry)
+            .setBaseImageName(baseImageRepository)
+            .setBaseImageTag(baseImageTag)
+            .setTargetServerUrl(registry)
+            .setTargetImageName(repository)
+            .setTargetTag(tag)
+            .setCredentialHelperName(credentialHelperName)
+            .setMainClass(mainClass)
+            .build();
+
+    Path cacheDirectory = Paths.get(project.getBuild().getDirectory(), CACHE_DIRECTORY_NAME);
+    if (!Files.exists(cacheDirectory)) {
+      try {
+        Files.createDirectory(cacheDirectory);
+
+      } catch (IOException ex) {
+        throw new MojoExecutionException("Could not create cache directory: " + cacheDirectory, ex);
+      }
+    }
+
+    try {
+      BuildImageSteps buildImageSteps =
+          new BuildImageSteps(buildConfiguration, sourceFilesConfiguration, cacheDirectory);
+      buildImageSteps.run();
+
+    } catch (RegistryUnauthorizedException ex) {
+      handleRegistryUnauthorizedException(ex);
+
+    } catch (IOException
+        | RegistryException
+        | CacheMetadataCorruptedException
+        | DuplicateLayerException
+        | LayerPropertyNotFoundException
+        | LayerCountMismatchException
+        | NonexistentDockerCredentialHelperException
+        | RegistryAuthenticationFailedException
+        | NonexistentServerUrlDockerCredentialHelperException ex) {
+      // TODO: Add more suggestions for various build failures.
+      throw new MojoExceptionBuilder(ex).build();
+    }
+  }
+
+  private SourceFilesConfiguration getSourceFilesConfiguration() throws MojoExecutionException {
+    try {
+      SourceFilesConfiguration sourceFilesConfiguration =
+          new MavenSourceFilesConfiguration(project);
+
+      // Logs the different source files used.
+      getLog().info("Dependencies:");
+      sourceFilesConfiguration
+          .getDependenciesFiles()
+          .forEach(dependencyFile -> getLog().info("Dependency: " + dependencyFile));
+
+      getLog().info("Resources:");
+      sourceFilesConfiguration
+          .getResourcesFiles()
+          .forEach(resourceFile -> getLog().info("Resource: " + resourceFile));
+
+      getLog().info("Classes:");
+      sourceFilesConfiguration
+          .getClassesFiles()
+          .forEach(classesFile -> getLog().info("Class: " + classesFile));
+
+      return sourceFilesConfiguration;
+
+    } catch (IOException ex) {
+      throw new MojoExecutionException("Obtaining project build output files failed", ex);
+    }
+  }
+
+  private void handleRegistryUnauthorizedException(RegistryUnauthorizedException ex)
+      throws MojoExecutionException {
+    MojoExceptionBuilder mojoExceptionBuilder = new MojoExceptionBuilder(ex);
```

Comment:
It's maybe not clear to me why we need this exception builder.

Could this class just have a helper to build to format the string and then just throw the exception in each conditional segment?